### PR TITLE
libvirt_vm: enablement to boot guest with Numa nodes

### PR DESCRIPTION
**Pull request by balamuruhans**
This patch enables option for virt-install to start the guest
with Numa nodes using unattended_install and execute other tests
by setting param "numa = yes" in cfg.

Signed-off-by: Balamuruhan S <bala24@linux.vnet.ibm.com>

### CODE REVIEW

File: virttest/libvirt_vm.py [hunk 1]
```diff
@@ -450,6 +450,44 @@ def add_smp(help_text, smp, maxvcpus=None, sockets=None,
                 cmd += ",threads=%s" % threads
             return cmd
 
+        def add_numa(vcpus, max_mem, numa_nodes):
+            """
+            Method to add Numa node to guest
+
+            :param vcpus: vcpus of guest
+            :param max_mem: max memory of guest
+
```
Comment (sathnaga) [inline virttest/libvirt_vm.py]: numa_nodes param missing

File: virttest/libvirt_vm.py [hunk 2]
```diff
@@ -450,6 +450,44 @@ def add_smp(help_text, smp, maxvcpus=None, sockets=None,
...
+            :return: appended numa parameter to virt-install cmd
+            """
+            cmd = " --cpu"
+            cell = "cell%s.cpus=%s,cell%s.id=%s,cell%s.memory=%s"
+            cells = ""
+
+            # we need atlease 1 vcpu for 1 numa node
```
Comment (sathnaga) [inline virttest/libvirt_vm.py]: s/atlease/at least

File: virttest/libvirt_vm.py [hunk 3]
```diff
@@ -450,6 +450,44 @@ def add_smp(help_text, smp, maxvcpus=None, sockets=None,
...
+                    cpus = cpus + cpus_balance
+                    memory = memory + memory_balance
+                if cpus == 1:
+                    cpu_str = "%s" % (cpu_start + (cpus - 1))
+                else:
+                    cpu_str = "%s-%s" % (cpu_start, cpu_start + (cpus - 1))
+                cpu_start += cpus
+                cells += "%s," % cell % (numa, cpu_str, numa, numa, numa, memory)
+            cmd += " %s" % cells
+            return cmd.strip(",")
```
Comment (sathnaga) [inline virttest/libvirt_vm.py]: +1

**Comment by balamuruhans** [comment]
@sathnaga Thankyou, I have made changes as suggested.

**Comment by balamuruhans** [comment]
```
# avocado run --vt-type libvirt --vt-config data/avocado-vt/backends/libvirt/cfg/lifecycle.cfg --vt-only-filter "RHEL.7.2.ppc64le scsi qcow2" --vt-no-filter "user network macvtap"
JOB ID     : 31a3677e83ded81e9e116947b93132ced19e6e91
JOB LOG    : /var/lib/libvirt/images/bala/avocado-fvt-wrapper/results/job-2017-07-21T11.09-31a3677/job.log
 (1/1) guest_import.qemu.qcow2.scsi.smp2.virtio_net.RHEL.7.2.ppc64le.powerkvm-qemu.unattended_install.import.import.default_install.aio_native: PASS (80.82 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 85.45 s
JOB HTML   : /var/lib/libvirt/images/bala/avocado-fvt-wrapper/results/job-2017-07-21T11.09-31a3677/html/results.html
```
**Log:**
```
2017-07-21 11:09:49,713 process          L0389 INFO | Running '/usr/bin/virt-install --connect=qemu:///system --hvm --accelerate --name 'virt-tests-vm1-Lifecycle' --machine pseries --memory=8388608 --vcpu=32,sockets=1,cores=4,threads=8 --cpu cell0.cpus=0-15,cell0.id=0,cell0.memory=4194304,cell1.cpus=16-31,cell1.id=1,cell1.memory=4194304 --import --nographics --serial pty --memballoon model=virtio --disk path=/var/lib/libvirt/images/bala/avocado-fvt-wrapper/data/avocado-vt/images/rhel72-ppc64le.qcow2,bus=scsi,size=10,format=qcow2 --network=bridge=virbr0,model=virtio,mac=52:54:00:e1:e2:e3 --noautoconsole'
2017-07-21 11:09:51,095 process          L0479 DEBUG| [stderr] WARNING  No operating system detected, VM performance may suffer. Specify an OS with --os-variant for optimal results.
2017-07-21 11:09:51,724 process          L0479 DEBUG| [stdout]
2017-07-21 11:09:51,725 process          L0479 DEBUG| [stdout] Starting install...
2017-07-21 11:09:51,725 process          L0479 DEBUG| [stdout]
2017-07-21 11:09:51,726 process          L0479 DEBUG| [stdout] Creating domain...                                          |    0 B  00:00
2017-07-21 11:09:51,731 process          L0479 DEBUG| [stdout] Domain creation completed.
2017-07-21 11:09:51,876 process          L0499 INFO | Command '/usr/bin/virt-install --connect=qemu:///system --hvm --accelerate --name 'virt-tests-vm1-Lifecycle' --machine pseries --memory=8388608 --vcpu=32,sockets=1,cores=4,threads=8 --cpu cell0.cpus=0-15,cell0.id=0,cell0.memory=4194304,cell1.cpus=16-31,cell1.id=1,cell1.memory=4194304 --import --nographics --serial pty --memballoon model=virtio --disk path=/var/lib/libvirt/images/bala/avocado-fvt-wrapper/data/avocado-vt/images/rhel72-ppc64le.qcow2,bus=scsi,size=10,format=qcow2 --network=bridge=virbr0,model=virtio,mac=52:54:00:e1:e2:e3 --noautoconsole' finished with 0 after 2.14882993698s
2017-07-21 11:09:51,876 utils_misc       L0659 DEBUG| waiting for domain virt-tests-vm1-Lifecycle to start (0.000012 secs)

```

**Pull request merged**